Source/CarthageKit/Git.swift

```diff
@@ -217,14 +227,24 @@ public func cloneSubmoduleInWorkingDirectory(_ submodule: Submodule, _ workingDi
 				}
 		}
 
-	return SignalProducer<(), CarthageError> { () -> Result<(), CarthageError> in
-		repositoryCheck("remove submodule checkout") {
-			try FileManager.default.removeItem(at: submoduleDirectoryURL)
+	let dependencyForSubmodule = Dependency.git(submodule.url)
+
+	return cloneOrFetchOnComplete(dependency: dependencyForSubmodule, preferHTTPS: preferHTTPS, commitish: submodule.sha)
+		.flatMap(.merge) { repositoryURL -> SignalProducer<String, CarthageError> in
+			return SignalProducer<(), CarthageError> { () -> Result<(), CarthageError> in
```

Comment:
It's not necessary to use `flatMap` with a custom `SignalProducer`. You can use `attemptMap` here instead.